### PR TITLE
Generate 100 particle sizes in RANDOM

### PR DESCRIPTION
**Pull request by Amber1995**

### CODE REVIEW

File: main.cc [hunk 1]
```diff
@@ -1 +1,12 @@
 #include <iostream>
+#include <stdlib.h>
```
Comment (kks32) [inline main.cc]:
```suggestion
#include <cstdlib>
```

File: main.cc [hunk 2]
```diff
@@ -1 +1,12 @@
...
+#include <time.h>
```
Comment (kks32) [inline main.cc]:
```suggestion
#include <ctime> 
```

File: main.cc [hunk 3]
```diff
@@ -1 +1,12 @@
 #include <iostream>
+#include <stdlib.h>
+#include <time.h> 
+
+int main(void)
```
Comment (kks32) [inline main.cc]:
```suggestion
int main() {
```

File: main.cc [hunk 4]
```diff
@@ -1 +1,12 @@
...
+{
```
Comment (kks32) [inline main.cc]:
```suggestion
```

File: main.cc [hunk 5]
```diff
@@ -1 +1,12 @@
...
+srand((unsigned)time(NULL));
```
Comment (kks32) [inline main.cc]: Use std::rand()

File: main.cc [hunk 6]
```diff
@@ -1 +1,12 @@
 #include <iostream>
+#include <stdlib.h>
+#include <time.h> 
+
+int main(void)
+{
+srand((unsigned)time(NULL)); 
+for(int i = 0; i < 100;i++ )
```
Comment (kks32) [inline main.cc]:
```suggestion
for(int i = 0; i < 100; ++i) 
```
Always preincreament

File: main.cc [hunk 7]
```diff
@@ -1 +1,12 @@
 #include <iostream>
+#include <stdlib.h>
+#include <time.h> 
+
+int main(void)
+{
+srand((unsigned)time(NULL)); 
+for(int i = 0; i < 100;i++ ) 
+    std::cout << rand() / double(RAND_MAX) << std::endl;
```
Comment (kks32) [inline main.cc]: I thought we wanted the radius to be between 2.0 and 4.0mm?

File: main.cc [hunk 8]
```diff
@@ -1 +1,12 @@
...
+return 0;
```
Comment (kks32) [inline main.cc]: We don't need return 0 when `main` gets to this point it will return true (0)

**Review by kks32 — CHANGES_REQUESTED** [review]
Could you please fix these

File: main.cc (hunk 5 above)
Comment (kks32) [reply main.cc]: https://en.cppreference.com/w/cpp/numeric/random/rand

File: main.cc [hunk 9]
```diff
@@ -1 +1,14 @@
+
+#include <cstdlib>
+#include <ctime>
 #include <iostream>
+
+int main() {
+  const double radius_a = 2;
+  double radius;
```
Comment (kks32) [inline main.cc]: You can remove that, this is not `c` where you have to declare everything up front

File: main.cc [hunk 10]
```diff
@@ -1 +1,14 @@
...
+  std::srand((unsigned)time(NULL));
+  for (int i = 0; i < 100; ++i) {
+    radius = radius_a + 2.0 * std::rand() / double(RAND_MAX);
```
Comment (kks32) [inline main.cc]: use `auto` 
```suggestion
    auto radius = radius_a + 2.0 * std::rand() / double(RAND_MAX);
```

**Review by kks32 — CHANGES_REQUESTED** [review]

**Review by kks32 — CHANGES_REQUESTED** [review]
Could you please create a vector of `radii` and append each new radius to that vector? 

Also could you please get the `n` number of `radii` needed from the user as an input. Just read it from commandline as an argument. Use `argc` and `argv` to parse commandline arguments.

**Comment by Amber1995** [comment]
Thanks for your suggestions！I will try to understand and make it. And as
soon as I finish it I will push it to GitHub.

See you tomorrow!

Qiuyu

Krishna Kumar <notifications@github.com>于2019年9月15日 周日下午6:02写道：

**Pull request merged**